src/test/regress/expected/failure_multi_row_insert.out

```diff
@@ -134,6 +134,14 @@ SELECT citus.mitmproxy('conn.onQuery(query="^INSERT").cancel(' || :pid || ')');
 
 INSERT INTO reference_table VALUES (1), (2), (3), (4);
 ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.onQuery(query="^INSERT").cancel(' || :pid || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+INSERT INTO distributed_table VALUES (1,1), (2,2), (3,3), (4,2), (5,2), (6,2), (7,2);
```

Comment:
And, what  happens  if you run the tests when you revert the  code changes? I'm curious

Reply:
Looking at it more this issue seems a bit theoretical. We normally just handle cancellations in CHECK_FOR_INTERRUPTS, which jumps out of SequentialRunDistributedExecution.

Reply:
Yeah after talking with Onder, it seems that this is theoretical because we would normally return in CHECK_FOR_INTERRUPTS and the problem would happen only if the execution is run and prior to the execution run HOLD_INTERRUPTS was called. This is not the case and it is not easy to reproduce. So adding the check in SequentialRunExecution is to prevent future problems, currently there is no problem with that because we do not call HOLD_INTERRUPTS prior to `RunDistributedExecution`.